bindings/python/cntk/ops/__init__.py

```diff
@@ -261,28 +261,87 @@ def labels_to_graph(labels, name=''):
+                     delay_constraint=-1, name=''):
     '''
-    Criterion node for training methods that rely on forward-backward Viterbi-like passes, e.g. Connectionist Temporal Classification (CTC) training
-    The node takes as the input the graph of labels, produced by the labels_to_graph operation that determines the exact forward/backward procedure.
+    Criterion node for training methods that rely on forward-backward Viterbi-like
+    passes, e.g. Connectionist Temporal Classification (CTC) training.
+    The node takes as the input the graph of labels, produced by the labels_to_graph
+    operation that determines the exact forward/backward procedure.
+
     Example:
-        graph = cntk.labels_to_graph(labels)
-        networkOut = model(features)
-        fb = C.forward_backward(graph, networkOut, 132)
+        labels_graph = cntk.labels_to_graph(labels_var)
+        network_out = model(input_var)
+        fb = C.forward_backward(labels_graph, network_out, 132)
+
+    Hint:
+        CNTK requires that both features and labels have the same time dimension.
+        This forward-backward code expects the duplication of these labels to
+        fulfill the requirements. The labels must be encoded as one-hot vectors,
+        where the populated index has values 1 or 2 (from now on referred to as
+        one-hot-1 and one-hot-2, respectively).
+
+        If the labels are frame-aligned, one could encode the vector as one-hot-2
+        if the label is in the frame boundary or as one-hot-1 if it is within the
+        frame. This is already done automatically by the HTKDeserializer.
+
+        If the labels are not frame-aligned, one could encode all labels as
+        one-hot-2, and then duplicate the last label as one-hot-1 until the
+        required time dimension is reached.
+
+        Example:
+            from sklearn.preprocessing import LabelBinarizer
+
+            lb = LabelBinarizer(pos_label=2).fit(range(6))
+            labels = lb.transform([0, 2, 0, 1, 3, 4]) # blank is the label 5
+            # labels = [[2,0,0,...,0,0], [0,0,2,...,0,0], ..., [0,0,0,...,2,0]]
+
+            # Retrieve the input time dimension
+            sequence_dim = input.shape[-2]
+
+            expanded_labels = np.zeros((sequence_dim, labels.shape[-1]))
+            # We first copy the original one-hot labels
+            expanded_labels[:len(labels)] = labels
+
+            # Then, we replicate the last label as one-hot-1 encoded
+            expanded_labels[len(labels):, labels[-1].argmax()] = 1
+            # expanded_labels = [[2,0,0,...,0,0], [0,0,2,...,0,0], ...,
+                                 [0,0,0,...,2,0], [0,0,0,...,1,0], ...,
+                                 [0,0,0,...,1,0]]
+
+            # We can define the model and the variables
+            input_var = sequence.input_variable((input.shape[-1]), name='input')
+            labels_var = sequence.input_variable((6), name='label')
+            # model = ...
```

Comment:
A little elaboration here would be good. So maybe a full comment saying that the model should be defined here.